src/store.ts

```diff
@@ -1,29 +1,98 @@
 import * as vsc from 'vscode';
+import * as Constants from './constants';
 import { UserStoryInfo, AzureClient, IterationInfo } from './utils/azure-client';
 import { Logger } from './utils/logger';
 import { Stopwatch } from './utils/stopwatch';
 import { Configuration } from './utils/config';
+import { TextProcessor } from './utils/textProcessor';
 
 export class SessionStore implements ISessionStore {
-	public currentIteration!: IterationInfo;
+	public currentIteration!: IterationInfo | undefined;
+	public customIteration!: IterationInfo | undefined;
+	public iterations?: IterationInfo[];
 	public userStories?: UserStoryInfo[] = undefined;
 
 	constructor(private azureClient: AzureClient, private config: Configuration, private logger: Logger) {
 	}
 
-	async ensureHasUserStories(): Promise<void> {
-		if (this.currentIteration && this.userStories !== undefined) {
+	async setIteration(): Promise<void> {
+		const editor = vsc.window.activeTextEditor;
+
+		if (editor) {
+			const lines = editor.document.getText().split(Constants.NewLineRegex);
+			const it = TextProcessor.getIteration(lines, 0);
+			if (!it) {
+				console.log('Iteration not specified - will default to @CurrentIteration');
+			} else {
+				this.customIteration = this.iterations!.find(x => x.id == it.id);
+				if (!this.customIteration) { return Promise.resolve(); }
+
+				this.logger.log(`Iteration set to ${this.customIteration.path.toString()}`);
+				vsc.window.setStatusBarMessage(`Iteration set to ${this.customIteration.path.toString()}`, 2000);
+			}
+		}		
+	
+		return Promise.resolve();
+	}
+
+	async ensureHasIterations (): Promise<void> {
+		if (!this.config.isValid) {
+			return Promise.reject("Missing URL or token in configuration");
+		}
+
+		try {
+			let total = Stopwatch.startNew();
+			this.iterations = await this.azureClient.getIterationsInfo();		
+			total.stop();
+
+			this.logger.log(`Iterations fetched in ${total.toString()} (3 requests)`);
+			vsc.window.setStatusBarMessage(`Iterations fetched in ${total.toString()} (3 requests)`, 2000);
+
+		} catch (err) {
+			this.logger.log(`[Error] ${err.message || err}`);
+			err.response && this.logger.log(`[Error] ${err.response.data.message}`);
+			return Promise.reject();
+		}
+
+		return Promise.resolve();
+	}
+
+
+	async ensureHasUserStories (): Promise<void> {
+		if (!this.customIteration && this.currentIteration && this.userStories !== undefined) {
 			return Promise.resolve();
 		}
 
+		// if (!this.currentIteration && this.customIteration && this.userStories !== undefined) {
+		// 	return Promise.resolve();
+		// }

```

Comment:
Yep, I'd tried to follow your pattern to avoid getting the stories more that once, but realised I wouldn't be able to change the iteration from default if that was left in, deleted.

Reply:
I think it's still broken. User stories are cached anyway so changing the iteration later still shows cached user stories.
For now I guess I will just ditch the caching and figure out a better solution later.